clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1932,6 +1939,13 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
   // class is entered.
   int StructBaseDepth = -1;
 
+  // Used to track FunctionDecl location in case if it is not available directly
+  // from method
+  SourceLocation FreeFunctionLoc;
+
+  // Used to check if FunctionDecl is a free function.
+  bool IsFreeFunction = false;
+
```

Comment:
I don't think this is necessary, you can check that the location is valid to understand that we have a free function.
```suggestion
```